src/definers.lisp

```diff
@@ -188,3 +188,84 @@
           (let* ((namespace (symbol-namespace ',name))
                  (hash-table (namespace-binding-table namespace)))
             (setf (gethash name hash-table) new-value)))))))
+
+;;; Definer forms
+
+(defun normalize-arglist (arglist)
+  "Makes sure the argument list contains an &REST parameter for &KEY and
+&OPTIONAL parameters"
+  (multiple-value-bind (required optional rest keywords allow-other-keys? aux
+                        keys?)
+      (parse-ordinary-lambda-list arglist
+                                  :normalize nil)
+    (let ((rest-arg (or rest (gensym "rest"))))
+      (values `(,@required
+                ,@(when optional
+                    `(&optional))
+                ,@optional
+                &rest ,rest-arg
+                ,@(when keys?
+                    `(&key))
+                ,@keywords
+                ,@(when allow-other-keys?
+                    `(&allow-other-keys))
+                ,@(when aux
+                    `(&aux))
+                ,@aux)
+              rest-arg
+              `(,@required ,@(mapcar #'ensure-car
+                                     optional))
+              (mapcar #'ensure-car
+                      keywords)))))
+
+(defun construct-function-definer-form (function name accessor)
+  (let ((g!name (gensym "name")))
+    (multiple-value-bind (arglist rest-argument normal-args k/o-args)
+        (normalize-arglist (if (symbolp function)
+                               (arglist function)
+                               (if (and (listp function)
+                                        (eq (first function)
+                                            'lambda))
+                                   (second function)
+                                   (error "Malformed function name"))))
```

Comment:
This kind of begs for a `cond` instead of the nested `if` and for extracting this value into a separate `arglist` variable for clarity.

Reply:
Yes, good point. It's done.